src/net/atlas/download.rs

```diff
@@ -88,8 +133,7 @@ impl AttachmentsDownloader {
                 }
 
                 let attachments_batch = self
```

Comment:
With the new interface methods, I think this expect can be avoided (and combined with the `is_empty` check) above while maintaining the same behavior:

```rust
let attachments_batch = match self.pop_next_ready_batch() {
    None => {
         warn!("Atlas: could not get a peer to sync with");
         // Nothing can be done!
         return Ok((vec![], vec![]));
    },
    Some(ready_batch) => ready_batch
}
```

While I think this is pretty much the exact same behavior, I think we're better off avoiding the `expect` if we can.